text/0000-document_all_features.md

```diff
@@ -0,0 +1,275 @@
+
+The basic decision has led to a substantial improvement in the currency of the documentation (which is now updated the same day as a new version is released). Moreover, it has spurred ongoing development of better tooling around documentation to manage these releases. Finally, at least in the RFC author's estimation, it has also led to a substantial increase in the overall quality of that documentation, possibly as a consequence of increasing the community involvement in the documentation process (including the formation of a documentation subteam).
+
+
+# Detailed design
+[design]: #detailed-design
+
+The basic process of developing new language features will remain largely the same as today. The required changes are two additions:
+
+- a new section in the RFC, "How do we teach this?" modeled on Ember's updated RFC process
+
+- a new requirement that the changes themselves be properly documented before being merged to stable
+
+
+## New RFC section: "How do we teach this?"
+[new-rfc-section]: #new-rfc-section-how-do-we-teach-this
+
+Following the example of Ember.js, we will add a new section to the RFC, just after **Detailed design**, titled **How do we teach this?** The section should to explain what changes need to be made to documentation, and if the feature substantially changes what would be considered the "best" way to solve a problem or is a fairly mainstream issue, discuss how it might be incorporated into _The Rust Programming Language_ and/or _Rust by Example_.
+
+Here is the Ember RFC section, with suggested substitutions:
+
+> # How We Teach This
+> What names and terminology work best for these concepts and why? How is this idea best presented? As a continuation of existing ~~Ember~~ **Rust** patterns, or as a wholly new one?
+>
+> Would the acceptance of this proposal mean ~~Ember guides~~ **_The Rust Programing Language_, _Rust by Example_, or the Rust Reference** must be re-organized or altered? Does it change how ~~Ember~~ **Rust** is taught to new users at any level?
+>
+> How should this feature be introduced and taught to existing ~~Ember~~ **Rust** users?
+
+We may also find it valuable to add other, more Rust-specific (or programming language- rather than framework-specific) verbiage there.
+
+For a great example of this in practice, see the (currently open) [Ember RFC: Module Unification], which includes several sections discussing conventions, tooling, concepts, and impacts on testing.
+
+[Ember RFC: Module Unification]: https://github.com/dgeb/rfcs/blob/module-unification/text/0000-module-unification.md#how-we-teach-this
+
+## New requirement to document changes before stabilizing
+
+Prior to approving pull requests that stabilize features, reviewers must verify that the document is properly documented in the following places:
+
+- language features:
+    - in the reference
+    - in _The Rust Programming Language_
+    - in _Rust by Example_
+- the standard library: in the `std` API docs
+
```

Comment:
Could we also add the originating RFC to this list of things that need to be updated?  Take for example the [closure RFC](https://github.com/rust-lang/rfcs/blob/master/text/0114-closures.md).  This RFC references things that doesn't seem to have stabilized like `FnShare` and receiver specifiers (which to be fair were removed from a different RFC).